crates/wit-component/tests/wit/floats.wat

```diff
@@ -0,0 +1,10 @@
+(component
+  (type (;0;) (func (param "x" float32)))
+  (type (;1;) (func (param "x" float64)))
+  (type (;2;) (func (result float32)))
+  (type (;3;) (func (result float64)))
+  (export "f32-param" (type 0))
+  (export "f64-param" (type 1))
+  (export "f32-result" (type 2))
+  (export "f64-result" (type 3))
```

Comment:
It was added to the spec as a way to give a name to a type such as a record or variant (or primitive type for a type alias). The spec used to have a "named" variant for an interface type def, but the component model authors thought exporting the type would be a more natural way to do that for a wasm spec.

Reply:
That's correct, it's primary for bindings generation.

Luke and I discussed using a custom section for it, but I believe the consensus was that bindings generation is important enough to not confine it to a potentially omittable custom section. And since type exports (and imports) will need to be a thing anyway eventually, it seemed like a natural way to name a type also for the purpose of bindings generation.

I think we should raise an issue on the spec if there's a better way to do this.

Reply:
I opened https://github.com/WebAssembly/component-model/issues/12 for this

Reply:
Resolving this comment as we can move it forward on the spec issue.